packages/codegen-doc/src/query.ts

```diff
@@ -18,17 +19,42 @@ export function findQuery(
     return undefined;
   }
 
-  const matchedNameAndType = context.queries.find(query => {
+  /** Select query if matching name and type */
+  const queryMatchingNameAndType = context.queries.find(query => {
     return (
       query.name.value.toLowerCase() === fieldName.toLowerCase() &&
       reduceTypeName(query.type) === type &&
       reduceListType(query.type) === listType
     );
   });
+  if (queryMatchingNameAndType) {
+    return queryMatchingNameAndType;
+  }
 
-  const matchedType = context.queries.find(query => {
+  /** Otherwise look for a matching type */
+  const queryMatchingType = context.queries.find(query => {
     return reduceTypeName(query.type) === type && reduceListType(query.type) === listType;
   });
+  if (!queryMatchingType) {
+    return undefined;
+  }
+
+  /** Get the matching object definition */
+  const responseFieldNames = context.objects
+    .find(obj => reduceTypeName(queryMatchingType.type) === obj.name.value)
+    ?.fields?.map(responseField => responseField.name.value);
+
+  if (!responseFieldNames) {
+    return undefined;
+  }
 
-  return matchedNameAndType ?? matchedType;
+  /** Ensure the fields are available on the object */
```

Comment:
The main change is contained to this file.

We were previously identifying a matching query for a model based on the returned type. 

This is not valid in the case where `ArchivedResponse` is queried with different required args than can be returned on the response object.

The fix finds the matching response object and confirms that the fields are available to be returned, to then be passed into this matching query at a later date, on the model.